examples/device/device.cpp

```diff
@@ -889,3 +919,226 @@ int CXX_QDMI_device_session_query_operation_property(
   }
   return QDMI_ERROR_NOTSUPPORTED;
 } /// [DOXYGEN FUNCTION END]
+
+int CXX_QDMI_device_session_query_telemetrysensor_property(
+    CXX_QDMI_Device_Session session, CXX_QDMI_TelemetrySensor telemetry_sensor,
+    QDMI_TelemetrySensor_Property prop, size_t size, void *value,
+    size_t *size_ret) {
+  if (session == nullptr || telemetry_sensor == nullptr ||
+      (value != nullptr && size == 0) ||
+      (prop >= QDMI_TELEMETRYSENSOR_PROPERTY_MAX &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM1 &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM2 &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM3 &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM4 &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM5)) {
+    return QDMI_ERROR_INVALIDARGUMENT;
+  }
+  ADD_STRING_PROPERTY(QDMI_TELEMETRYSENSOR_PROPERTY_ID,
+                      telemetry_sensor->id.c_str(), prop, size, value, size_ret)
+  ADD_STRING_PROPERTY(QDMI_TELEMETRYSENSOR_PROPERTY_UNIT,
+                      telemetry_sensor->unit.c_str(), prop, size, value,
+                      size_ret)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_TELEMETRYSENSOR_PROPERTY_SAMPLINGRATE,
+                            uint64_t, telemetry_sensor->sampling_rate, prop,
+                            size, value, size_ret)
+  return QDMI_ERROR_NOTSUPPORTED;
+}
+
+int CXX_QDMI_device_session_create_device_telemetrysensor_query(
+    CXX_QDMI_Device_Session session,
+    CXX_QDMI_Device_TelemetrySensor_Query *query) {
+
+  if (session == nullptr || query == nullptr) {
+    return QDMI_ERROR_INVALIDARGUMENT;
+  }
+  if (session->status != CXX_QDMI_DEVICE_SESSION_STATUS::INITIALIZED) {
+    return QDMI_ERROR_BADSTATE;
+  }
+  *query = new CXX_QDMI_Device_TelemetrySensor_Query_impl_d();
+  (*query)->start_time = std::chrono::system_clock::now();
+  (*query)->end_time = std::chrono::system_clock::now();
+  (*query)->status = QDMI_TELEMETRYSENSOR_QUERY_STATUS_CREATED;
+
+  return QDMI_SUCCESS;
+}
+
+int CXX_QDMI_device_telemetrysensor_query_set_parameter(
+    CXX_QDMI_Device_TelemetrySensor_Query query,
+    QDMI_Device_TelemetrySensor_Query_Parameter param, size_t size,
+    const void *value) {
+
+  if (query == nullptr || (value != nullptr && size == 0) ||
+      (param >= QDMI_DEVICE_TELEMETRYSENSOR_QUERY_PARAMETER_MAX &&
+       param != QDMI_DEVICE_TELEMETRYSENSOR_QUERY_PARAMETER_CUSTOM1 &&
+       param != QDMI_DEVICE_TELEMETRYSENSOR_QUERY_PARAMETER_CUSTOM2 &&
+       param != QDMI_DEVICE_TELEMETRYSENSOR_QUERY_PARAMETER_CUSTOM3 &&
+       param != QDMI_DEVICE_TELEMETRYSENSOR_QUERY_PARAMETER_CUSTOM4 &&
+       param != QDMI_DEVICE_TELEMETRYSENSOR_QUERY_PARAMETER_CUSTOM5) ||
+      value == nullptr || (size != sizeof(decltype(query->start_time)))) {
```

Comment:
```suggestion
      value == nullptr) {
```
The check `(size != sizeof(decltype(query->start_time)))` must be placed somewhere differently because also other information than of the start_time's type can be returned, e.g., a sensor pointer. This check should be handled in the respective cases.

Reply:
Corrected.